File: api_checker/lib/api_checker/endpoints/endpoint.ex
defmodule ApiChecker.Endpoints.Endpoint do
  use Ecto.Schema
  import Ecto.Changeset

  require Logger

  schema "endpoints" do
    field :active, :boolean, default: false
    field :name, :string
    field :url, :string
    field :check_interval_seconds, :integer
    field :notification_email, :string
    field :notification_slack_webhook, :string

    belongs_to :user, ApiChecker.Accounts.User

    has_many :check_results, ApiChecker.Endpoints.CheckResult
    timestamps(type: :utc_datetime)
  end

  @doc false
  def changeset(endpoint, attrs) do
    changeset =
      endpoint
      |> cast(attrs, [:url, :check_interval_seconds, :active, :name, :notification_email, :notification_slack_webhook, :user_id])

    changeset
    |> validate_required([:url, :check_interval_seconds, :active, :name, :user_id])
  end
end

File: frontend/src/types/endpoint.ts
export interface Endpoint {
    id: number;
    url: string;
    name: string;
    active: boolean;
    notification_email: string | null;
    check_interval_seconds: number;
    inserted_at: string;
    updated_at: string;
  }
  
  export interface CheckResult {
    id: number;
    status_code: number;
    response_time_ms: number;
    endpoint_id: number;
    checked_at: string;
  }
  